.github/workflows/workflow.yml

```diff
@@ -15,59 +15,49 @@ jobs:
           - windows-latest
         # Please keep the list in sync with the minimal version of OCaml in
         # dune-project, dune.opam.template and bootstrap.ml
-        ocaml-version:
-          - 4.13.1
+        #
+        # We don't run tests on all versions of the Windows environment and on
+        # 4.02.x and 4.07.x in other environments
+        ocaml-compiler:
+          - 4.13.x
         skip_test:
           - false
         include:
-          - ocaml-version: 4.12.1
+          - ocaml-compiler: 4.12.x
             os: ubuntu-latest
             skip_test: true
-          - ocaml-version: 4.11.1
+          - ocaml-compiler: 4.11.x
             os: ubuntu-latest
             skip_test: true
-          - ocaml-version: 4.10.0
+          - ocaml-compiler: 4.10.x
             os: ubuntu-latest
             skip_test: true
-          - ocaml-version: 4.09.1
+          - ocaml-compiler: 4.09.x
             os: ubuntu-latest
             skip_test: true
-          - ocaml-version: 4.08.1
+          - ocaml-compiler: 4.08.x
             os: ubuntu-latest
             skip_test: true
-          - ocaml-version: 4.04.2
+          - ocaml-compiler: 4.04.x
             os: ubuntu-latest
             skip_test: true
             configurator: true
-          - ocaml-version: 4.02.3
+          - ocaml-compiler: 4.02.x
             os: ubuntu-latest
             skip_test: true
 
-    # We don't run tests on all versions of the Windows environment and on
-    # 4.02.3 and 4.07.1 in other environments
-    env:
-      # Note that this environment variable is recognized as a string by GitHub
-      # Actions, so the following doesn't actually work:
-      #
-      # if: env.SKIP_TEST
-      #
-      # To skip an action, the following should be used:
-      # if: ${{ env.SKIP_TEST != "false" }}
-      SKIP_TEST: ${{ matrix.skip_test }}
-      OCAML_VERSION: ${{ matrix.ocaml-version }}
-      OS: ${{ matrix.os }}
-      CONFIGURATOR: ${{ matrix.configurator }}
-
     runs-on: ${{ matrix.os }}
 
     steps:
       - name: Checkout code
         uses: actions/checkout@v2
 
-      - name: Use OCaml ${{ matrix.ocaml-version }}
-        uses: ocaml/setup-ocaml@v1
+      - name: Use OCaml ${{ matrix.ocaml-compiler }}
+        uses: ocaml/setup-ocaml@v2
         with:
-          ocaml-version: ${{ matrix.ocaml-version }}
+          ocaml-compiler: ${{ matrix.ocaml-compiler }}
+          dune-cache: true
```

Comment:
I think that we can start without this feature for now, especially since it seems to be causing issues on windows.